lom/Makefile

```diff
@@ -61,97 +64,164 @@ format_code:
 
 # Define the file coverage extraction function
 define extract_file_coverage
-	@echo "*******************************************************"
-	@echo "Extracting file coverages:"
-	@file_coverage="$$(sed -n 's/.*<option value="[^"]\+">\([^<]\+\).* (\([^)]\+\%.*\)<\/option>.*/\1 (\2)/p' $(1))"; 
-	echo "$${file_coverage}"
-	@echo "*******************************************************"
-	@echo ""
-	@echo ""
+#!/bin/bash
+echo "====================================================================================="
+echo -e "Detail file coverage from $$1 :"
+echo "_____________________________________________________________________________________"
+file_coverage="$$(sed -n 's/.*<option value="[^"]\+">\([^<]\+\).* (\([^)]\+\%.*\)<\/option>.*/\1 (\2)/p' "$$1")"; 
+echo "$${file_coverage}"
+echo "====================================================================================="
 endef
+export extract_file_coverage
+
+add-extract_file_coverage: 
+	$(shell mkdir -p $(BUILD_DIR))
+	$(shell rm -f  $(BUILD_DIR)/extract_file_coverage.sh)
+	$(shell echo >> $(BUILD_DIR)/extract_file_coverage.sh)
+	@echo "$$extract_file_coverage" >> $(BUILD_DIR)/extract_file_coverage.sh
+	$(shell chmod 777 $(BUILD_DIR)/extract_file_coverage.sh 2>/dev/null || :)
+
+# Clean up the script file
+clean-extract_file_coverage:
+	rm -f $(BUILD_DIR)/extract_file_coverage.sh
+
+# Define the aggregate file coverage extraction function
+define aggregate_file_coverage
+#!/bin/bash
+echo "====================================================================================="
+echo -e "Aggregate file coverages : "
+echo "_____________________________________________________________________________________"
+for file in $(TEST_DIR)/*.html; do
+    if [ -f "$$file" ]; then
+        file_coverage=$$(sed -n 's/.*<option value="[^"]\+">\([^<]\+\).* (\([^)]\+\%.*\)<\/option>.*/\1 (\2)/p' "$$file")
+        echo "$$file_coverage"
+    fi
+done
+echo "_____________________________________________________________________________________"
+endef
+export aggregate_file_coverage
+
+# Print all the source fiules coverages at one place
+add-aggregate_file_coverage: 
+	$(shell mkdir -p $(BUILD_DIR))
+	$(shell rm -f  $(BUILD_DIR)/aggregate_file_coverage.sh)
+	$(shell echo >> $(BUILD_DIR)/aggregate_file_coverage.sh)
```

Comment:
curious! Why we need this ?

Reply:
removed line 79 and did single redirect later